src/cljs/athens/page.cljs

```diff
@@ -70,14 +70,37 @@
        [:div {:style {:margin-left 20}}
         [render-blocks (:block/uid @node)]]])))
 
+(defn title [title]
+  (let [s (reagent/atom {:editing false
+                         :current-title title
+                         :new-title title})
+        save! (fn [] (dispatch [:node/rename (:current-title @s) (:new-title @s)])
+                     (swap! s assoc :editing false))
+        cancel! (fn [] (swap! s #(-> %
+                                     (assoc :editing false)
+                                     (assoc :new-title (:current-title @s)))))]
+    (fn [title]
+      (if (:editing @s)
+        [:input {:value (:new-title @s)
+                 :auto-focus true
```

Comment:
I think you need to do auto-focus because otherwise you are losing focus, right? How does auto-focus work? Is it perfect?

Reply:
Indeed, if I don't add this attribute, the user would have to click twice on the h2 node to actually start editing. I think a limitation of auto-focus is that there can only be one element on the page with this attribute. I'm not sure what happens when there is more than one element on the page.

An alternative could be to make this a form-3 Reagent component, so I get a reference to the DOM node. Then I can call `.focus` on it.

Reply:
Ah I see. I guess adding `:auto-focus` makes sense then. I was thinking you might have added it in relation to my suggestion of "losing focus" in my comment below.